packages/material-ui/src/Breadcrumbs/BreadcrumbCollapsed.js

```diff
@@ -1,52 +1,92 @@
 import * as React from 'react';
 import PropTypes from 'prop-types';
-import withStyles from '../styles/withStyles';
+import { unstable_composeClasses as composeClasses } from '@material-ui/unstyled';
+import experimentalStyled from '../styles/experimentalStyled';
 import { emphasize } from '../styles/colorManipulator';
 import MoreHorizIcon from '../internal/svg-icons/MoreHoriz';
 import ButtonBase from '../ButtonBase';
+import { getBreadcrumbCollapsedUtilityClass } from './breadcrumbCollapsedClasses';
 
-const styles = (theme) => ({
-  button: {
-    display: 'flex',
-    marginLeft: theme.spacing(0.5),
-    marginRight: theme.spacing(0.5),
-    backgroundColor: theme.palette.grey[100],
-    color: theme.palette.grey[700],
-    borderRadius: 2,
-    '&:hover, &:focus': {
-      backgroundColor: theme.palette.grey[200],
-    },
-    '&:active': {
-      boxShadow: theme.shadows[0],
-      backgroundColor: emphasize(theme.palette.grey[200], 0.12),
-    },
+const useUtilityClasses = (styleProps) => {
+  const { classes } = styleProps;
+
+  const slots = {
+    button: ['button'],
+    icon: ['icon'],
+  };
+
+  return composeClasses(slots, getBreadcrumbCollapsedUtilityClass, classes);
+};
+
+const BreadcrumbCollapsedButton = experimentalStyled(
+  ButtonBase,
+  {},
+  {
+    name: 'PrivateBreadcrumbCollapsed',
+    slot: 'Button',
+  },
+)(({ theme }) => ({
+  display: 'flex',
+  marginLeft: theme.spacing(0.5),
+  marginRight: theme.spacing(0.5),
+  ...(theme.palette.type === 'light'
+    ? { backgroundColor: theme.palette.grey[100], color: theme.palette.grey[700] }
+    : { backgroundColor: theme.palette.grey[700], color: theme.palette.grey[100] }),
+  borderRadius: 2,
+  '&:hover, &:focus': {
+    ...(theme.palette.type === 'light'
+      ? { backgroundColor: theme.palette.grey[200] }
+      : { backgroundColor: theme.palette.grey[600] }),
   },
-  icon: {
-    width: 24,
-    height: 16,
+  '&:active': {
+    boxShadow: theme.shadows[0],
+    ...(theme.palette.type === 'light'
+      ? { backgroundColor: emphasize(theme.palette.grey[200], 0.12) }
+      : { backgroundColor: emphasize(theme.palette.grey[600], 0.12) }),
   },
+}));
+
+const BreadcrumbCollapsedIcon = experimentalStyled(
+  MoreHorizIcon,
+  {},
+  {
+    name: 'PrivateBreadcrumbCollapsed',
+    slot: 'Icon',
+  },
+)({
+  width: 24,
+  height: 16,
 });
 
 /**
  * @ignore - internal component.
  */
-function BreadcrumbCollapsed(props) {
-  const { classes, ...other } = props;
+const BreadcrumbCollapsed = React.forwardRef(function BreadcrumbCollapsed(props, ref) {
+  const { ...other } = props;
+
+  const styleProps = { ...props };
```

Comment:
```suggestion
  const styleProps = props;
```

We should be able to do this.